bpf/lib/nodeport.h

```diff
@@ -1091,15 +1091,16 @@ static __always_inline int rev_nodeport_lb6(struct __ctx_buff *ctx, __u32 *ifind
 			if (eth_store_saddr(ctx, fib_params.smac, 0) < 0)
 				return DROP_WRITE_ERROR;
 		}
-	} else {
-		if (!bpf_skip_recirculation(ctx)) {
-			ctx_skip_nodeport_set(ctx);
-			ep_tail_call(ctx, CILIUM_CALL_IPV6_FROM_NETDEV);
-			return DROP_MISSED_TAIL_CALL;
-		}
+
+		return CTX_ACT_REDIRECT;
 	}
 
-	return CTX_ACT_REDIRECT;
+	if (bpf_skip_recirculation(ctx))
+		return DROP_NAT_NO_MAPPING;
```

Comment:
If I understand correctly the impact of this commit, in case of a mismatch between `egress_ct_state->node_port` and `ingress_ct_state` (i.e., first is true but second doesn't exist for some reason), we would silently drop the packet because we would call `bpf_redirect` with an incorrect ifindex. Now, we will emit a Cilium drop event.

So clearly a good change. And not a bugfix because we don't expect this mismatch to ever happen today (correct me if wrong @borkmann).

Nevertheless, it is a change in behavior, so could we explicitly mention it in the commit message? That is, mention the user-visible impact.

Reply:
> Nevertheless, it is a change in behavior, so could we explicitly mention it in the commit message?

At this stage I don't expect it to have any user-visible impact (ie. it's totally an "impossible to hit" condition). But we do need it in the following patches to have a clean `goto out` path.